tools/define_sanity/check.py

```diff
@@ -0,0 +1,76 @@
+import fnmatch
+import glob
+import os
+import re
+import sys
+
+def green(text):
+    return "\033[32m" + str(text) + "\033[0m"
+
+def red(text):
+    return "\033[31m" + str(text) + "\033[0m"
+
+def post_error(define_name, file, github_error_style):
+    if github_error_style:
+        print(f"::error file={file},title=Define Sanity::{define_name} is defined locally in {file} but not undefined locally!")
+    else:
+        print(red(f"- Error parsing {file}: {define_name} is defined locally in {file} but not undefined locally!"))
+
+parent_directory = "code/**/*.dm"
+# This files/directories are expected to have "global" defines, so they must be exempt from this check.
+excluded_files = [
+    #  Wildcard directories, all files are expected to be exempt.
+    "code/__DEFINES/*.dm",
+    "code/__HELPERS/*.dm",
+    "code/_globalvars/*.dm",
+    # Singular files.
+    "code/__byond_version_compat.dm",
+    "code/_compile_options.dm",
+    # TGS files come from another repository so lets not touch them.
+    "code/modules/tgs/v3210/commands.dm",
+    "code/modules/tgs/v4/api.dm",
+    "code/modules/tgs/v5/_defines.dm",
+    "code/modules/tgs/core/_definitions.dm",
+    "code/modules/unit_tests/_unit_tests.dm",
+    # In an ideal world, anything below this line shouldn't be the way it is and moved to the __DEFINES directory. This is not an ideal world.
+    # If you disagree, feel free to either refactor them out of this list or move them to the "Singular Files" section.
```

Comment:
I don't agree.

If a define is only used by its folder, then a `_defines.dm` makes complete sense.

Reply:
it doesn't reduce surface area as much as it shifts it elsewhere but i agree on the premise that the defines folder is getting rather bulky so encapsulation makes sense

Reply:
Code kept in one folder is definitely less surface area than having it also litter a file in __DEFINES